administrator/components/com_csp/Model/ReportsModel.php

```diff
@@ -106,6 +106,15 @@ protected function getListQuery()
 			->select('*')
 			->from($db->quoteName('#__csp', 'a'));
 
+		// Filter by client
+		$client = (string) $this->getState('filter.client');
+
+		if (!empty($client))
+		{
+			$query->where($db->quoteName('a.client') . ' = :client')
+				->bind(':client', $client, ParameterType::STRING);
```

Comment:
```suggestion
				->bind(':client', $client);
```
don't need to specify if it is a STRING  (default)

Reply:
I saw that parameter value in lots of places. How do I know when it is the default

Reply:
https://github.com/joomla-framework/database/blob/4d61aef4b76880c381e44e7df16d206130c8d6d6/src/Query/PreparableInterface.php#L38

Reply:
That doesn't really answer my question - unless you are saying the 60 occurrences of ::string are also wrong

Reply:
Different methods have different defaults. Nic's link is confusing because it's pointing to 1.x version but in J4 we're using 2.0. Relevant methods for 2.0 are:

https://github.com/joomla-framework/database/blob/ca0d0ca7b117189d7c0240cd05ac932908cf411d/src/DatabaseQuery.php#L1991
https://github.com/joomla-framework/database/blob/2.0-dev/src/DatabaseQuery.php#L2094
https://github.com/joomla-framework/database/blob/ca0d0ca7b117189d7c0240cd05ac932908cf411d/src/DatabaseQuery.php#L1877

Reply:
So is this line of code wrong as well
https://github.com/joomla/joomla-cms/blob/87512f237b4ad6a50f72d77dff1198657d06ec2c/administrator/components/com_installer/Helper/InstallerHelper.php#L259-L262

Reply:
259, 260, 262  are not wrong  simply not necessary  cause of default

Reply:
all clear now -  thanks